tests/mypy/test_directives.yml

```diff
@@ -0,0 +1,20 @@
+- case: test_directive_decorator
+  main: |
+    import strawberry
+    from strawberry.directive import DirectiveLocation
+
+    @strawberry.directive(
+        locations=[DirectiveLocation.FRAGMENT_DEFINITION],
+        description="description.",
+    )
+    def make_int(value: str) -> int:
+        """description."""
+        try:
+            return int(value)
+        except ValueError:
+            return 0
+
+    reveal_type(make_int)
+
+  out: |
+    main:15: note: Revealed type is "strawberry.directive.StrawberryDirective[builtins.int]"
```

Comment:
this might be for another PR, but I wonder if we should return the function as the type, it is true that we wrap it but that should just be an internal thing (similar to what we do for scalars, types, etc)

I don't remember why we wrap the function instead of monkeypatching it.

I'll make a note to check this in future, I don't want to block this PR 😊